controllers/humiocluster_controller.go

```diff
@@ -2040,10 +2041,14 @@ func (r *HumioClusterReconciler) ingressesMatch(ingress *networkingv1.Ingress, d
 	return true
 }
 
-func (r *HumioClusterReconciler) ensurePodsExist(ctx context.Context, hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool) (reconcile.Result, error) {
+func (r *HumioClusterReconciler) ensurePodsExist(ctx context.Context, hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool, req ctrl.Request) (reconcile.Result, error) {
 	// Ensure we have pods for the defined NodeCount.
-	// If scaling down, we will handle the extra/obsolete pods later.
-	foundPodList, err := kubernetes.ListPods(ctx, r, hnp.GetNamespace(), hnp.GetNodePoolLabels())
+
+	// Exclude pods that are currently being evicted --> Ensures K8s keeps track of the pods waiting for eviction and doesn't remove pods continuously
+	labelsToMatch := hnp.GetNodePoolLabels()
+	labelsToMatch[kubernetes.PodMarkedForDataEviction] = "false"
```

Comment:
Are we using labels or annotations to manage this?

Reply:
This seems to conflict with other areas that are adding the annotation.

Reply:
See below